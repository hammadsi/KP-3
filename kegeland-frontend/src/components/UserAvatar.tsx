import { Wrap, WrapItem, Avatar, Text } from '@chakra-ui/react';
import { AiOutlineUser } from 'react-icons/ai';

import { User } from '../state/ducks/auth/auth.interface';
import { ThemeMode } from '../types';
import { renderName } from '../utils/renderName';

type UserAvatarProps = {
  user: User;
  mode?: ThemeMode;
};

const UserAvatar: React.FC<UserAvatarProps> = ({ user, mode }) => {
  const isDark = mode === 'dark';
  return (
    <Wrap>
      <WrapItem alignItems="center">
        <Avatar
          size="xs"
          bg="primary.600"
          color="white"
          name={renderName(user.name)}
          icon={<AiOutlineUser />}
        />
        <Text
          paddingLeft={1}
          fontWeight={'semibold'}
          color={isDark ? 'gray.200' : 'gray.700'}
        >
          {renderName(user.name)}
        </Text>
      </WrapItem>
    </Wrap>
  );
};

UserAvatar.defaultProps = {
  mode: 'dark',
};

export default UserAvatar;
