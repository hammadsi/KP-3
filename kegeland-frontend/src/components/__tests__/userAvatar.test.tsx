import renderer from 'react-test-renderer';

import { userMock } from '../mocks/user.mock';
import UserAvatar from '../UserAvatar';

describe('Test user avatar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<UserAvatar user={userMock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
