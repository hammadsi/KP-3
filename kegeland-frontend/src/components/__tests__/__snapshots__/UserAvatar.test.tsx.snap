// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test user avatar renders correctly 1`] = `
<div
  className="chakra-wrap css-i6bazn"
>
  <ul
    className="chakra-wrap__list css-wah4g8"
  >
    <li
      className="chakra-wrap__listitem css-70qvj9"
    >
      <span
        className="chakra-avatar css-ew7jl2"
      >
        <div
          aria-label="ola Nordmann"
          className="chakra-avatar__initials css-0"
          role="img"
        >
          oN
        </div>
      </span>
      <p
        className="chakra-text css-zyu9tc"
      >
        ola Nordmann
      </p>
    </li>
  </ul>
</div>
`;
